sdk/storage/azure-storage-file-datalake/src/main/java/com/azure/storage/file/datalake/DataLakeFileClient.java

```diff
@@ -276,10 +278,14 @@ public void read(OutputStream stream) {
      */
     public FileReadResponse readWithResponse(OutputStream stream, FileRange range, DownloadRetryOptions options,
         DataLakeRequestConditions requestConditions, boolean getRangeContentMd5, Duration timeout, Context context) {
-        BlobDownloadResponse response = blockBlobClient.downloadWithResponse(stream, Transforms.toBlobRange(range),
-            Transforms.toBlobDownloadRetryOptions(options), Transforms.toBlobRequestConditions(requestConditions),
-            getRangeContentMd5, timeout, context);
-        return Transforms.toFileReadResponse(response);
+        try {
+            BlobDownloadResponse response = blockBlobClient.downloadWithResponse(stream, Transforms.toBlobRange(range),
```

Comment:
Any chance to have the sync `readWithResponse` call into the async `readWithResponse` in a similar fashion that Blobs handles sync download calling into async download? That way the try/catch convert exception won't need to be written everywhere.

Reply:
The reason we cant do this is because we have no way to pass the context (since we dont have the package private method)